test/applications/partial_lotka_volterra.jl

```diff
@@ -0,0 +1,165 @@
+    end
+    false
+end
+@info "Initial loss $(loss(p)[1])"
+@info "Train neural network until converged"
+res1 = DiffEqFlux.sciml_train(loss, p, ADAM(0.01), cb=callback, maxiters = 100)
+@info "Finished initial training with loss $(losses[end])"
+res2 = DiffEqFlux.sciml_train(loss, res1.minimizer, BFGS(initial_stepnorm=0.01), cb=callback, maxiters = 10000)
+@info "Finished extended training with loss $(losses[end])"
+
+# Check for the convergence of loss
+# Necessary for result
+@test losses[end] <= 1e-2
+
+# Plot the data and the approximation
+NNsolution = predict(res2.minimizer)
+
+@test norm(NNsolution .- solution, 2) < 0.1
+
+Y = ann(noisy_data, res2.minimizer)
+X = noisy_data
+# Create a Basis
+@variables u[1:2]
+# Lots of polynomials
+polys = Operation[1]
+for i ∈ 1:5
+    push!(polys, u[1]^i)
+    push!(polys, u[2]^i)
+    for j ∈ i:5
+        if i != j
+            push!(polys, (u[1]^i)*(u[2]^j))
+            push!(polys, u[2]^i*u[1]^i)
+        end
+    end
+end
+
+# And some other stuff
+h = [cos.(u)...; sin.(u)...; polys...]
+basis = Basis(h, u)
+
+# Create an optimizer for the SINDY problem
+opt = SR3()
+# Create the thresholds which should be used in the search process
+λ = exp10.(-10:0.05:-0.5)
+# Target function to choose the results from; x = L0 of coefficients and L2-Error of the model
+function eval_target(x)
+    y = similar(x)
+    if iszero(x[1])
+        y[1] = convert(eltype(x), Inf)
+    end
+    y[2] = x[2]
+    return y
+end
+
+alg = GoalProgramming(x->norm(x, 2), eval_target)
+@info "Start sindy regression with unknown threshold"
+# Test on uode derivative data
+Ψ = SInDy(X[:, 2:end], Y[:, 2:end], basis, λ,  opt = opt, maxiter = 10000, normalize = true, denoise = true, alg = alg) # Succeed
+p̂ = parameters(Ψ)
+@info "Build initial guess system"
+# The parameters are a bit off, so we reiterate another sindy term to get closer to the ground truth
+# Create function
+unknown_sys = ODESystem(Ψ)
+unknown_eq = ODEFunction(unknown_sys)
+# Just the equations
+b = Basis((u, p, t)->unknown_eq(u, ones(size(p̂)), t), u)
+# Test on uode derivative data
+@info "Refine the guess"
+Ψ = SInDy(X[:, 2:end], Y[:, 2:end],b, opt = SR3(0.1), maxiter = 1000) # Succeed
+p̂ = parameters(Ψ)
+
+
+@test isapprox(abs.(p̂), p_[2:3], atol = 9e-2)
+
+# The parameters are a bit off, so we reiterate another sindy term to get closer to the ground truth
+# Create function
+unknown_sys = ODESystem(Ψ)
+unknown_eq = ODEFunction(unknown_sys)
+
+
+# Build a ODE for the estimated system
+function approx(du, u, p, t)
+    # Add SInDy Term
+    α, δ, β, γ = p
+    z = unknown_eq(u, [β; γ], t)
+    du[1] = α*u[1] + z[1]
+    du[2] = -δ*u[2] + z[2]
+end
+@info "Simulate system"
+# Create the approximated problem and solution
+ps = [p_[[1,4]]; p̂]
+a_prob = ODEProblem(approx, u0, tspan, ps)
+a_solution = solve(a_prob, Vern7(), abstol=1e-12, reltol=1e-12, saveat = solution.t)
+@test norm(a_solution-solution, 2) < 5e-1
+@test norm(a_solution-solution, Inf) < 1.5e-1
```

Comment:
can the learned basis be tested to be the quadratic terms?

Reply:
Yes